src/app-layer-ssl.c

```diff
@@ -1431,6 +1431,10 @@ static int EnsureRecordSpace(SSLStateConnp *curr_connp, const uint8_t * const in
         SCLogDebug("cert_len unknown still, create small buffer to start");
         certs_len = 256;
     }
+    // Limit in a first time allocation for very large certificates
+    if (certs_len > 0x10000 && certs_len > curr_connp->trec_pos + input_len) {
```

Comment:
should we first check if current input will set the limited value?

Also please use a macro with a clear name, a decimal value and a comment to explain the default

Reply:
> should we first check if current input will set the limited value?

I am not sure I understand your question.
This is a first check that current input will set the limited value...

That is if not `certs_len > curr_connp->trec_pos + input_len` that means that the current input will cross the 65k boundary, and thus, we do not limit ourselves